<img src="https://lifg.australian.museum/Image/9uTxr6do.jpeg?version=full" alt="Zdi" width="300"/>

# ATLAS: _Zenarchopterus dispar_ lcWGS

Following the [ATLAS pipeline](https://github.com/philippinespire/pire_lcwgs_data_processing/tree/main/scripts/ATLAS_wahab) for Zdi.

Done by Gianna Mazzei (December 2024).

---

## ATLAS

<details><summary>1. Pre-processing </summary>

### 1. Pre-processing

Make an ATLAS directory in the same directory that holds your GenErode directory:
```
[hpc-0373@wahab-01 pire_zenarchopterus_dispar_lcwgs]$ mkdir ATLAS_Zdi
```
Now, make sure you have generated GERP scores:
```
# gerp scores
[hpc-0373@wahab-01 pire_zenarchopterus_dispar_lcwgs]$ cd GenErode_Zdi_4/results/gerp/
[hpc-0373@wahab-01 gerp]$ ls *gz
reference.genbank.Zdi.20k.ancestral.rates.gz
```
And the proper .bam files needed for ATLAS:

historical:
```
cd /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/GenErode_Zdi_4/results/historical/mapping/reference.genbank.Zdi.20k/
[hpc-0373@wahab-01 reference.denovoSSL.Pbb]$ ls *merged.rmdup.merged.realn.bam -1 | wc -l
47

# compare to input:
[hpc-0373@wahab-01 reference.genbank.Zdi.20k]$ cd ../../../../historical/
[hpc-0373@wahab-01 historical]$ ls -1 | cut -c 1-12 | uniq | wc -l
47
```
modern:
```
cd /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/GenErode_Zdi_4/results/modern/mapping/reference.genbank.Zdi.20k/
[hpc-0373@wahab-01 reference.denovoSSL.Pbb]$ ls *merged.rmdup.merged.realn.bam -1 | wc -l
64

# compare to input:
[hpc-0373@wahab-01 reference.genbank.Zdi.20k]$ cd ../../../../modern/
[hpc-0373@wahab-01 modern]$ ls -1 | cut -c 1-12 | uniq | wc -l
64
```
---
</details>

<details><summary>2. Binning GERP Scores</summary>
  
### Binning GERP Scores

To generate bed files based on binned GERP scores for ATLAS, you will have to process the file generated by GenErode. 

So far we have used three bins (sites with scores >1.5, sites with scores between 1.0 and 1.5, and sites with scores between 0.5 and 1).

Now get sites with scores in each particular bin: 
```
# sites with scores greater than or equal to 1.5
[hpc-0373@wahab-01 pire_zenarchopterus_dispar_lcwgs]$ gunzip -c GenErode_Zdi_4/results/gerp/reference.genbank.Zdi.20k.ancestral.rates.gz | awk -v OFS='\t' '($4 >= 1.5) {print $1,$2,$2}' > ATLAS_Zdi/gerp_15.bed

# sites with scores greater than or equal to 1, but less than 1.5
[hpc-0373@wahab-01 pire_zenarchopterus_dispar_lcwgs]$ gunzip -c GenErode_Zdi_4/results/gerp/reference.genbank.Zdi.20k.ancestral.rates.gz | awk -v OFS='\t' '($4 >= 1 && $4 < 1.5) {print $1,$2,$2}' > ATLAS_Zdi/gerp_1_15.bed

# sites with scores greater than or equal to 0.5, but less than 1
[hpc-0373@wahab-01 pire_zenarchopterus_dispar_lcwgs]$ gunzip -c GenErode_Zdi_4/results/gerp/reference.genbank.Zdi.20k.ancestral.rates.gz | awk -v OFS='\t' '($4 >= 0.5 && $4 < 1) {print $1,$2,$2}' > ATLAS_Zdi/gerp_05_1.bed
```
Now check those gerp*.bed files to see the number of sites. We are aiming for ~5 million total.
```
[hpc-0373@wahab-01 ATLAS_Zdi]$ wc -l gerp_15.bed
66391 gerp_15.bed
[hpc-0373@wahab-01 ATLAS_Zdi]$ wc -l gerp_1_15.bed
1764823 gerp_1_15.bed
[hpc-0373@wahab-01 ATLAS_Zdi]$ wc -l gerp_05_1.bed
4988494 gerp_05_1.bed
```
Total is 6,819,708 so we can move forward with these bins.

Now, merge adjacent sites into regions using bedtools.
```
[hpc-0373@wahab-01 ATLAS_Zdi]$ module load bedtools

[hpc-0373@wahab-01 ATLAS_Zdi]$ crun bedtools merge -i gerp_15.bed > gerp_15.merge.bed 
[hpc-0373@wahab-01 ATLAS_Zdi]$ crun bedtools merge -i gerp_1_15.bed > gerp_1_15.merge.bed 
[hpc-0373@wahab-01 ATLAS_Zdi]$ crun bedtools merge -i gerp_05_1.bed > gerp_05_1.merge.bed 
```

Remove singleton sites from merged .bed files.
```
[hpc-0373@wahab-01 ATLAS_Zdi]$ awk -v OFS='\t' '!($2==$3)' < gerp_15.merge.bed > gerp_15.merge.nosingle.bed 
[hpc-0373@wahab-01 ATLAS_Zdi]$ awk -v OFS='\t' '!($2==$3)' < gerp_1_15.merge.bed > gerp_1_15.merge.nosingle.bed 
[hpc-0373@wahab-01 ATLAS_Zdi]$ awk -v OFS='\t' '!($2==$3)' < gerp_05_1.merge.bed > gerp_05_1.merge.nosingle.bed 
```
---
</details>

<details><summary>3. Get & Edit Scripts</summary>

### 3. Get & Edit Scripts

Copy scripts to ATLAS dir:
```
[hpc-0373@wahab-01 ATLAS_Zdi]$ cp /home/e1garcia/shotgun_PIRE/pire_lcwgs_data_processing/scripts/ATLAS_wahab/*.sbatch .
[hpc-0373@wahab-01 ATLAS_Zdi]$ cp /home/e1garcia/shotgun_PIRE/pire_lcwgs_data_processing/scripts/ATLAS_wahab/*.bash .
```
**For Step 4:**

Edit the `atlas_recal_readuntilbeds_array.sbatch` script to reflect the number of .bed files you want to use and their names. 
```
[hpc-0373@wahab-01 ATLAS_Zdi]$ cat -n  atlas_recal_readuntilbeds_array.sbatch
# last line:
    34	crun.atlas atlas --task estimateErrors --bam ${sample_name}.merged.rmdup.merged.realn.bam --fasta ${REFLOC} --regions gerp_05_1.merge.nosingle.bed,gerp_1_15.merge.nosingle.bed,gerp_15.merge.nosingle.bed --readUpToDepth 10
```
Edit the `atlas_recal_readuntilbeds_array.bash` script to reflect the path for the `atlas_recal_readuntilbeds_array.sbatch` script
```
[hpc-0373@wahab-01 ATLAS_Zdi]$ cat -n atlas_recal_readuntilbeds_array.bash
# last line:
    20	       /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/ATLAS_Zdi/atlas_recal_readuntilbeds_array.sbatch ${BAMDIR} ${REFLOC} ${OUTDIR})
```

**For Step 5:**

Edit the `atlas_theta_albrecal_array.bash` script to reflect the path for the `atlas_theta_albrecal_array.sbatch` script.
```
[hpc-0373@wahab-01 ATLAS_Zdi]$ cat -n atlas_theta_albrecal_array.bash
# last line:
    19	       /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/ATLAS_Zdi/atlas_theta_albrecal_array.sbatch ${BAMDIR} ${OUTDIR})
```
---
</details>

<details><summary>4. Run Recalibration</summary>

### 4. Run Recalibration

Make an output directory, and copy the bed files to it:
```
[hpc-0373@wahab-01 ATLAS_Zdi]$ mkdir recal_output
[hpc-0373@wahab-01 ATLAS_Zdi]$ cp *.bed recal_output/
```

Run recalibration on **historical** files:
```
# format:
# bash atlas_recal_readuntilbeds_array.bash [directory with .bam files] [reference genome location w/ filename] [output directory]

[hpc-0373@wahab-01 ATLAS_Zdi]$ bash atlas_recal_readuntilbeds_array.bash /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/GenErode_Zdi_4/results/historical/mapping/reference.genbank.Zdi.20k/ /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/GenErode_Zdi_4/reference/reference.genbank.Zdi.20k.fasta /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/ATLAS_Zdi/recal_output/
```
Make sure all files were created, comparing the input to the output:
```
## input
ls /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/GenErode_Zdi_4/results/historical/mapping/reference.genbank.Zdi.20k/*merged.rmdup.merged.realn.bam -1 | wc -l
47

## output
ls /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/ATLAS_Zdi/recal_output/ZdiA*.merged.rmdup.merged.realn.bam | wc -l
47

ls /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/ATLAS_Zdi/recal_output/ZdiA*.merged.rmdup.merged.realn_RGInfo.json | wc -l
47
```

Run recalibration on **modern** files:
```
# format:
# bash atlas_recal_readuntilbeds_array.bash [directory with .bam files] [reference genome location w/ filename] [output directory]

[hpc-0373@wahab-01 ATLAS_Zdi]$ bash atlas_recal_readuntilbeds_array.bash /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/GenErode_Zdi_4/results/modern/mapping/reference.genbank.Zdi.20k/ /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/GenErode_Zdi_4/reference/reference.genbank.Zdi.20k.fasta /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/ATLAS_Zdi/recal_output/
```
Make sure all files were created, comparing the input to the output:
```
## input
ls /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/GenErode_Zdi_4/results/modern/mapping/reference.genbank.Zdi.20k/*merged.rmdup.merged.realn.bam -1 | wc -l
64

## output
ls /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/ATLAS_Zdi/recal_output/ZdiC*.merged.rmdup.merged.realn.bam | wc -l
64

ls /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/ATLAS_Zdi/recal_output/ZdiC*.merged.rmdup.merged.realn_RGInfo.json | wc -l
64
```
---
</details>

<details><summary>5. Estimate Theta</summary>

### 5. Estimate Theta

Now, we will estimate theta for both historic and modern samples using the `*merged.rmdup.merged.realn.bam` and `*.merged.rmdup.merged.realn_RGInfo.json` files in our output directory and perform downsampling.

Make an output directory:
```
[hpc-0373@wahab-01 ATLAS_Zdi]$ mkdir theta_output
```
Run `atlas_theta_albrecal_array.bash`:
```
# format:
# bash atlas_theta_albrecal_array.bash [directory with recalibrated output files] [output directory]

[hpc-0356@wahab-01 ATLAS_Zdi]$ bash atlas_theta_albrecal_array.bash /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/ATLAS_Zdi/recal_output/ /archive/carpenterlab/pire/pire_zenarchopterus_dispar_lcwgs/ATLAS_Zdi/theta_output/
```
---
</details>


